gc/gc-base/src/main/java/org/projectnessie/gc/base/GCParams.java

```diff
@@ -89,14 +79,31 @@ default double getBloomFilterFpp() {
 
   @Value.Check
   default void validate() {
-    Integer taskCount = getSparkPartitionsCount();
-    if (taskCount != null && taskCount <= 0) {
-      throw new IllegalArgumentException("taskCount has invalid value: " + taskCount);
+    Instant now = Instant.now();
```

Comment:
All these changes are unrelated to the scope of the PR.

Reply:
I really wanted to have validations that ensure these timestamp objects are not from the future. 
It can be a separate PR, but don't want too many PR for nit stuffs.

Reply:
Never ever make assumptions about the wall clock.
Yes, I do want to have this check moved out, because your assumption that wall clock is monotonically increasing is not correct and Instants in the future make sense.

Reply:
ok. moved out now.